src/std/stxparam.ss

```diff
@@ -31,6 +31,27 @@ package: std
    (defsyntax id
      (make-syntax-parameter key: (gensym 'id) default: default))))
 
+(defsyntax (defsyntax-parameter* stx)
+  (def (defparam macro param (errmsg "Bad syntax"))
+    `(begin
+       (defsyntax-parameter ,param #f)
+       (defsyntax (,macro stx)
+         (if (identifier? stx)
```

Comment:
we might want to allow applicative uses as well, as it is it can't be used as a function.

Reply:
But we have cases (like in message.ss) where we know it can't be a function.

Reply:
Do you want this change inside this PR or another?

Reply:
it's ok for now, but let's keep it in mind for when we have something that is a function parameter;
perhaps add a comment.